src/main/java/flaxbeard/steamcraft/block/BlockBoiler.java

```diff
@@ -237,13 +237,14 @@ public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer p
     	TileEntityBoiler tileEntity = (TileEntityBoiler) world.getTileEntity(x, y, z);
 
         boolean isClient = !world.isRemote;
-        
-	    if (!FluidHelper.playerIsHoldingWaterContainer(player) && isClient && tileEntity != null) {
-    		player.openGui(Steamcraft.instance, 0, world, x, y, z);
-        } else {
-        	FluidHelper.fillTankFromHeldItem(player, tileEntity.myTank);
+        if (tileEntity != null) {
+        	if (!FluidHelper.playerIsHoldingWaterContainer(player) && isClient && tileEntity != null) {
```

Comment:
Kind of. Your null check should be there, but the one below it should be removed, as the statement under the `else` block tries to access a member in tileEntity.